app/models/katello/host/subscription_facet.rb

```diff
@@ -242,14 +242,21 @@ def remove_subscriptions(pools_with_quantities)
 
       def self.find_host(facts, organization)
         host_name = propose_existing_hostname(facts)
-        hosts = ::Host.unscoped.where(:name => host_name)
+        hosts = ::Host.unscoped.distinct.left_outer_joins(:fact_values)
+                .where("#{::Host.table_name}.name = ? OR (#{FactValue.table_name}.fact_name_id = ?
+		AND #{FactValue.table_name}.value = ?)", host_name,
+                FactName.where(name: 'dmi::system::uuid').pluck(:id), facts['dmi.system.uuid'])
+
+        return nil if hosts.empty?
 
-        return nil if hosts.empty? #no host exists
         if hosts.where("organization_id = #{organization.id} OR organization_id is NULL").empty? #not in the correct org
           #TODO: http://projects.theforeman.org/issues/11532
           fail _("Host with name %{host_name} is currently registered to a different org, please migrate host to %{org_name}.") %
                    {:org_name => organization.name, :host_name => host_name }
         end
+
+        fail _("Could not associate this host to an existing profile since multiple were found. Please remove outdated content hosts.") if hosts.size > 1
```

Comment:
If we keep the failure mode here, would like some opinions on the messaging!

Reply:
Personally, I do think the failure message is appropriate to keep. I think this one should be reworded for clarity? "Multiple profiles exist for this host. Please remove outdated content hosts." Or something along those lines? Just needs to be more succinct methinks.

Reply:
Yeah, mine was super wordy. I like yours!

Reply:
could you include something like the host ids or hostnames, to make it easier on the user to identify?   
Hostnames might leak information, but a) the user has to be authed with at least an activation key, b) the only way they would conflict is if it already had the same hostname, or had the UUID of the existing system which seems strong enough to depend on.  So i'm leaning towards names

Reply:
@cfouant @jlsherrill 

Updated the message to include matching hostnames - looks something like this:

```
Multiple profiles found. Consider removing host12, host13 which match this host.
```